archive/2025-11-01-agents-rule-of-two-a-practical-approach-to-ai-agent-security.md

```diff
@@ -0,0 +1,321 @@
+
+[BACK](https://ai.meta.com/blog/practical-ai-agent-security/# "Go up one level")
+
+- [Explore Meta AI](https://ai.meta.com/meta-ai/)
+- [Get Meta AI](https://ai.meta.com/get-meta-ai/)
+- [Meta AI Studio](https://ai.meta.com/ai-studio/)
+
+- [Overview](https://ai.meta.com/research/)
+- [Projects](https://ai.meta.com/research/#projects)
+- [Research Areas](https://ai.meta.com/research/#research-areas)
+- [People](https://ai.meta.com/results/?content_types[0]=person)
+
+- [Overview](https://ai.meta.com/about/)
+- [Open Source](https://ai.meta.com/opensourceai/)
+- [Careers](https://www.metacareers.com/)
+
+Clear
+
+- Clear
+
+- [Meta AI\\
+\\
+>](https://ai.meta.com/blog/practical-ai-agent-security/#)
+
+- [AI Research\\
+\\
+>](https://ai.meta.com/blog/practical-ai-agent-security/#)
+
+- [The Latest](https://ai.meta.com/blog/)
+- [About\\
+\\
+>](https://ai.meta.com/blog/practical-ai-agent-security/#)
+
+- [Get Llama](https://www.llama.com/?utm_source=ai_meta_site&utm_medium=web&utm_content=AI_nav&utm_campaign=09252025_moment)
+
+[Try Meta AI](https://www.meta.ai/?utm_source=ai_meta_site&utm_medium=web&utm_content=AI_nav&utm_campaign=09252025_moment)
+
+FEATURED
+
+# Agents Rule of Two: A Practical Approach to AI Agent Security
+
+October 31, 2025•
+
+14 minute read
+
+![](https://scontent.xx.fbcdn.net/v/t39.2365-6/569630476_853480637349455_6406966111987148057_n.png?_nc_cat=101&ccb=1-7&_nc_sid=e280be&_nc_ohc=DWiUu9ko2RgQ7kNvwHs_Qiu&_nc_oc=Admbat1Qrj2EWrjg2_i3V0pw23hyDQYiaRg9P6FBjy4TJDd8OSQm6komx4t6WtdRJmI&_nc_zt=14&_nc_ht=scontent.xx&_nc_gid=8o7n7oJtzLiGltDHrDz9jA&oh=00_Afen1aBYD-rOyxImyf7kWx_YA4Qs1vp937sx7VyHQfXh_Q&oe=6920003B)
+
+Imagine a personal AI agent, Email-Bot, that’s designed to help you manage your inbox. In order to provide value and operate effectively, Email-Bot might need to:
+
+- Access unread email contents from various senders to provide helpful summaries
+- Read through your existing email inbox to keep track of any important updates, reminders, or context
+- Send replies or follow-up emails on your behalf
+
+While the automated email assistant can be of great help, this hypothetical bot can also demonstrate how AI agents are introducing novel risks. Notably, one of the biggest challenges for the industry is that of agents’ susceptibility to [prompt injection](https://en.wikipedia.org/wiki/Prompt_injection).
+
+Prompt injection is a fundamental, unsolved weakness in all LLMs. With prompt injection, certain types of untrustworthy strings or pieces of data — when passed into an AI agent’s context window — can cause unintended consequences, such as ignoring the instructions and safety guidelines provided by the developer or executing unauthorized tasks. This vulnerability could be enough for an attacker to take control of the agent and cause harm to the AI agent’s user.
```

Comment:
Great to see this written out explicitly. Just a few months ago labs and app devs were still trying to say that prompt injection can be fixed. Some at Anthropic and OAI still do.